core/src/main/java/io/github/thibaultbee/streampack/internal/gl/FullFrameRect.kt

```diff
@@ -91,8 +91,10 @@ class FullFrameRect(var program: Texture2DProgram) {
      *
      * The appropriate EGL context must be current.
      */
-    fun changeProgram(program: Texture2DProgram) {
-        this.program.release()
+    fun changeProgram(program: Texture2DProgram, releasePrevious: Boolean = true) {
```

Comment:
in my implementation, each encoding path has its own fullframerect (since the resolutions might differ) but they all share the same program. So I dont want them to nuke the program out from under eachother